### PR TITLE
NWR Private bonus, and close Canadian Pacific properly

### PR DESCRIPTION
**Pull request by jenf**
Implements remaining 1882 privates

### CODE REVIEW

File: assets/app/view/game/round/operating.rb [hunk 1]
```diff
@@ -20,6 +20,7 @@ def render
           @step = round.active_step
           corporation = round.current_entity
           @current_actions = round.actions_for(corporation)
+          corporation = corporation.owner if corporation.company?
```
Comment (tobymao) [inline assets/app/view/game/round/operating.rb:23]: maybe change corporation to entity

File: lib/engine/action/remove_token.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require_relative 'base'
+
+module Engine
+  module Action
+    class RemoveToken < Base
+      attr_reader :city, :slot, :token
```
Comment (tobymao) [inline lib/engine/action/remove_token.rb:8]: where is token set?

File: lib/engine/step/g_1882/nwr_track_bonus.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+# frozen_string_literal: true
+
+module Engine
+  module Step
+    module G1882
+      module NwrTrackBonus
+        def tile_nwr?(tile)
+          nwr_name = 'NWR'
```
Comment (tobymao) [inline lib/engine/step/g_1882/nwr_track_bonus.rb:8]: just do this inline icon.name == 'NWR'

File: lib/engine/step/g_1882/special_nwr.rb [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+        def pass_description
+          'NWR: Skip (Track)'
+        end
+
+        def active_entities
+          [@entity]
+        end
+
+        def blocks?
+          @state != nil
```
Comment (tobymao) [inline lib/engine/step/g_1882/special_nwr.rb:57]: just do @state

File: lib/engine/action/remove_token.rb (hunk 2 above)
Comment (jenf) [reply lib/engine/action/remove_token.rb:8]: it's not :)

**Pull request merged**